File: src/main/java/RemoveFavoriteAction.java
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.PlatformDataKeys;
import com.kubicz.mavenexecutor.window.MavenExecutorService;
import com.kubicz.mavenexecutor.window.MavenExecutorToolWindow;
import org.jetbrains.idea.maven.project.MavenProjectsManager;
import org.jetbrains.idea.maven.utils.actions.MavenActionUtil;

import javax.swing.*;
import java.awt.*;


public class RemoveFavoriteAction extends AnAction {

    public RemoveFavoriteAction() {
        super("");
    }

    public void actionPerformed(AnActionEvent event) {
        MavenExecutorService settingsService = MavenExecutorService.getInstance(event.getProject());

        JButton favoriteButton = (JButton)event.getData(PlatformDataKeys.CONTEXT_COMPONENT);

        settingsService.removeFavoriteSettings(favoriteButton.getText());

        MavenExecutorToolWindow.getInstance(event.getProject()).updateFavorite();
    }

}

File: src/main/java/RemoveFavoriteAction.kt
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.PlatformDataKeys
import org.kubicz.mavenexecutor.view.window.MavenExecutorService
import org.kubicz.mavenexecutor.view.window.MavenExecutorToolWindow
import javax.swing.JButton


class RemoveFavoriteAction : AnAction("") {

    override fun actionPerformed(event: AnActionEvent) {
        val settingsService = MavenExecutorService.getInstance(event.project!!)

        val favoriteButton = event.getData(PlatformDataKeys.CONTEXT_COMPONENT) as JButton

        settingsService.removeFavoriteSettings(favoriteButton.text)

        MavenExecutorToolWindow.getInstance(event.project!!).updateFavorite()
    }

}